cmd/lint.go

```diff
@@ -152,24 +152,30 @@ func Lint(isDebug *bool) *cli.Command {
 			}
 
 			var result *lint.PipelineAnalysisResult
+			var errr error
 			if asset == "" {
 				linter := lint.NewLinter(path.GetPipelinePaths, DefaultPipelineBuilder, rules, logger)
 				logger.Debugf("running %d rules for pipeline validation", len(rules))
 				infoPrinter.Printf("Validating pipelines in '%s' for '%s' environment...\n", rootPath, cm.SelectedEnvironmentName)
-				result, err = linter.Lint(rootPath, pipelineDefinitionFile)
+				result, errr = linter.Lint(rootPath, pipelineDefinitionFile)
 			} else {
 				rules = lint.FilterRulesByLevel(rules, lint.LevelAsset)
 				logger.Debugf("running %d rules for asset-only validation", len(rules))
 				linter := lint.NewLinter(path.GetPipelinePaths, DefaultPipelineBuilder, rules, logger)
-				result, err = linter.LintAsset(rootPath, pipelineDefinitionFile, asset)
+				result, errr = linter.LintAsset(rootPath, pipelineDefinitionFile, asset)
 			}
 
 			printer := lint.Printer{RootCheckPath: rootPath}
+			if errr != nil || result == nil {
+				printError(err, c.String("output"), "An error occurred")
+				return cli.Exit("", 1)
+			}
+
```

Comment:
this would print the error even when it is supposed to print JSON with the `output` flag, I believe this should be placed after the if-statement right below this, line 174.